### DISCUSSION TITLE
4 state model - moveHMM/momentuHMM - randomized initial parameter values

### DISCUSSION THREAD

**Discussion by RyanBoswarthick** [TheoMichelot/moveHMM#36] [Q&A] [answered]
Hello to all :) 
I'm currently working on some Northern gannet GPS data, and I've been trying to do a 4 state model with the momentuHMM package.. but with **randomized parameter values.**

I've followed Theo's script in part 5 of the great document named "_a short guide to choose initial parameter values for the estimation in moveHMM_". It works for the 2 state model, but i've been trying to do the same process for a 4 state model, but I can't produce a fully working script.

So that's why I'm here, after many attempts :) 

If you have any suggestions, tips, to fit a 4 state model with the "runif" method, I would be happy to hear them.

Thank a lot in advance,

Ryan.

**Comment by TheoMichelot** [comment, marked as answer]
Hi Ryan,

You mention that you don't have a working script. What exactly is the problem? Does the model fitting fail some of the times? Or all of the time? Or does it work, but return strange results?

It is always more difficult to pick starting values for models with many states, because there are many more parameters to estimate. As with simpler models, the general idea is to use plausible values of the parameters as starting values, but "plausible" is harder to define.

One strategy, which can help to some extent, is to increase complexity incrementally. That is, start with a 2-state model, and find good starting values for this one. Then, use the estimates from the 2-state model to come up with informed guesses for the starting values of a 3-state model, and so on. This should be easier than to start with 4 states. In momentuHMM, you could have a look at the function `getPar0`, which can help selecting starting values for a complex model, given a simpler model.

If you know how to select one plausible value for each model parameter (e.g., using the strategy outlined above), then this should help you come up with a plausible range of values, which is what you need to pass to `runif` to generate random starting values. The range you simulate from could for example be centred on your "best guess". Choosing the width of the range is a difficult trade-off: if it is too narrow, then you might not explore the parameter space enough, but if it is too wide, then you will probably be generating some very implausible parameters (or combinations of parameters). There is no general method to do this, but it would probably involve looking at histograms of the data variables.

I hope that this helps, please let me know if you have any further questions.

Théo

**Reply by RyanBoswarthick** [reply]
Hello Theo, thank you for your complete answer.

I've managed to solve errors for both models, the 2-state and the 4-state model, that now work (but not with the "runif" method).
The "getPar0" seems to be a good option to help me get suitable starting values.

I will do my best and use all of your tips, I hope it will work. 

Thank you so much,

Best regards,

Ryan.
